bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java

```diff
@@ -56,12 +58,37 @@ public AbstractActionSequenceElement<EntryLevelSystemCall> evaluateDataFlow(List
                 .flatMap(it -> it.getVariableCharacterisation_VariableUsage()
                         .stream())
                     .collect(Collectors.toList());
+    	
+    	if (this.isCalling()) {
+        	checkParameter(variableCharacterisations);
+        }
+    	
 
         PCMDataCharacteristicsCalculator characteristicsCalculator = new PCMDataCharacteristicsCalculator(new ArrayList<>(variables), nodeCharacteristics);
         variableCharacterisations.stream()
             .forEach(it -> characteristicsCalculator.evaluate(it));
        return new CallingUserActionSequenceElement(this, characteristicsCalculator.getCalculatedCharacteristics(), nodeCharacteristics);
     }
+    
+    private void checkParameter(List<VariableCharacterisation> variableCharacterisations) {
+    	List<String> parameter = 
+    			this.getElement().getOperationSignature__EntryLevelSystemCall().getParameters__OperationSignature().stream()
+    			.map(it -> it.getParameterName())
+    			.toList();
+    	
+    	List<String> referencedParameter =
```

Comment:
The second part of this looks a lot like code duplication. Of course, the first part has to be specific to ExternalCalls / EntryLevelSystemCalls, but the second is the same, right?

Reply:
I've moved the duplicate code into the `PCMDataCharacteristicsCalculator` to remove the code duplication here